lib/iris/tests/unit/analysis/test_PERCENTILE.py

```diff
@@ -52,9 +52,9 @@ def check_percentile_calc(
             self.assertFalse(is_lazy)
 
         if approx:
-            self.assertArrayAlmostEqual(actual, expected)
+            self.assertMaskedArrayAlmostEqual(actual, expected)
         else:
-            self.assertArrayEqual(actual, expected)
+            self.assertMaskedArrayEqual(actual, expected)
```

Comment:
Adding this has caused test failures (e.g. `FAIL: test_masked_2d_multi (unit.analysis.test_PERCENTILE.Test_aggregate)`) @rcomer 

I think they're legitimate (see line 131) because we've got an `mdtol` of 0.1 that is exceeded by the 0.33 that we're sending in, so we should just be getting a masked array back.

What was your intention on this test when you were in this code previously? Should it just return a masked array, or do we want to check that the results come back right?

Reply:
I went for the best of both worlds, by masking half the columns then checking the array was part masked.

Imagine my surprise when one of those tests is now failing because when we do the lazy version, masking half the input points masks half the percentages in the output...

Reply:
I don't actually have any memory of this!  It looks like I did it deliberately when modifying the lazy mdtol handling: https://github.com/rcomer/iris/commit/46397cbd5c6e990d7f9cee97a8049fc9c5b41c00  So I really should have noticed it wasn't having the desired effect 😔

Maybe there should be two versions of this - with and without the `mdtol` keyword?

Reply:
> I really should have noticed it wasn't having the desired effect

We only started getting this failure since I added the `assertMaskedArrayEqual`, which you spotted as reviewer here when I missed it reviewing yours...

Reply:
> > I really should have noticed it wasn't having the desired effect
> 
> We only started getting this failure since I added the `assertMaskedArrayEqual`, which you spotted as reviewer here when I missed it reviewing yours...

It's all fine as long as noone else notices.  Good thing we're not doing all this development in public...

Reply:
Note to Monday @wjbenfold - line 594-599 of `iris/analysis/__init__.py` is decidedly suspect

Reply:
OK I just checked out f41fbbb37c54a093b8c7ca8ad698fb7a58ac3a26, which is the last commit on main before #3901.  I changed that version of the test to

```python
    def test_masked_2d_multi(self):
        shape = (3, 10)
        data = ma.arange(np.prod(shape)).reshape(shape)
        data[1, ::2] = ma.masked
        percent = np.array([10, 50, 70, 80])
        actual = PERCENTILE.aggregate(data, axis=0, percent=percent, mdtol=0.1)
        self.assertTupleEqual(actual.shape, (shape[-1], percent.size))
        expected = np.tile(ma.arange(shape[-1]), percent.size)
        expected = expected.reshape(percent.size, shape[-1]).T
        expected = expected + (percent / 10 * 2)
        expected[::2] = ma.masked
        self.assertMaskedArrayAlmostEqual(actual, expected)
```

which I think is equivalent to what you now have here.  It failed comparing the masks.  So this seems to have been broken before we ever touched it.

Reply:
https://github.com/SciTools/iris/pull/4755/commits/2a42c1891c05313ab07f705317dd30b1c12301fa